generator/src/Google/Service/Generator/Method.php

```diff
@@ -0,0 +1,162 @@
+
+    private $paramsP = [];
+    private $paramsA = [];
+    public $hasPostBody = false;
+  
+    public function __construct($key, &$method)
+    {
+        $this->id = $method['id'];
+        $this->name = $key;
+        $this->path = $method['path'];
+        $this->httpMethod = $method['httpMethod'];
+        if (isset($method['description'])) {
+            $this->description = $method['description'];
+        }
+
+        if (isset($method['parameters'])) {
+            $this->emptyParameters = false;
+
+            $parameters = [];
+            if (isset($method['parameterOrder'])) {
+                foreach ($method['parameterOrder'] as $movedKey) {
+                    $parameters[$movedKey] = $method['parameters'][$movedKey];
+                    unset($method['parameters'][$movedKey]);
+                }
+            }
+            ksort($method['parameters']);
+            $this->parameters = array_merge($parameters, $method['parameters']);
+
+            foreach ($this->parameters as $k => $v) {
+                if (isset($v['required'])) {
+                    $var_name = '$'. lcfirst(StringUtilities::ucstrip($k));
+                    $this->paramsP[] = $var_name;
+                    $this->paramsA[] = "'$k' => $var_name";
+                    $this->phpdocParams[] = Method::getPhpDocParam($var_name, $v);
+                } else {
+                    $var_name = lcfirst(StringUtilities::ucstrip($k));
+                    $this->phpdocOptParams[] = Method::getPhpDocParam($var_name, $v);
+                }
+            }
+        }
+
+        if (isset($method['request']) && $method['httpMethod'] != 'GET') {
+            $this->requestRef = $method['request']['$ref'];
+            $this->hasPostBody = true;
+        }
+
+        if (isset($method['response'])) {
+            $this->responseRef = $method['response']['$ref'];
+        }
+    }
+
+    public function getParamsParams($modelclassname)
+    {
+        $params = $this->paramsP;
+        if ($this->hasPostBody) {
+            $params[] = "$modelclassname \$postBody";
+        }
+        $params[] = '$optParams = array()';
+        return implode(", ", $params);
+    }
+
+    public function getParamsArray()
+    {
+        $params = $this->paramsA;
+        if ($this->hasPostBody) {
+            $params[] = "'postBody' => \$postBody";
+        }
+        return implode(", ", $params);
+    }
+
+    public function getDescription()
+    {
+        $tmparr = explode('.', $this->id);
+        array_shift($tmparr);
+        $desc = '('. implode('.', $tmparr) . ')';
+        if ($this->description != '') {
+            $desc = "$this->description $desc";
+        }
+        return StringUtilities::commentWordwrap($desc);
+    }
+
+    public static function getPhpDocParam($name, $node)
+    {
+        if (isset($node['required'])) {
+            $token = '@param';
+
+            if (isset($node['repeated'])) { // TODO Why is this only for required parameters???
```

Comment:
Ternary? :)

also, a friendly ping on the todo - is there anything left to discover?

Reply:
☑️